packages/slate-react/src/components/editable.tsx

```diff
@@ -380,18 +380,39 @@ export const Editable = (props: EditableProps) => {
           const [node, offset] = ReactEditor.toDOMPoint(editor, anchor)
           const anchorNode = node.parentElement?.closest('a')
 
-          if (anchorNode && ReactEditor.hasDOMNode(editor, anchorNode)) {
-            const { document } = ReactEditor.getWindow(editor)
+          const window = ReactEditor.getWindow(editor)
 
+          if (
+            native &&
+            anchorNode &&
+            ReactEditor.hasDOMNode(editor, anchorNode)
+          ) {
             // Find the last text node inside the anchor.
-            const lastText = document
+            const lastText = window.document
               .createTreeWalker(anchorNode, NodeFilter.SHOW_TEXT)
               .lastChild() as DOMText | null
 
             if (lastText === node && lastText.textContent?.length === offset) {
               native = false
             }
           }
+
+          // Chrome has issues with the presence of tab characters inside elements with whiteSpace = 'pre'
+          // causing abnormal insert behavior: https://bugs.chromium.org/p/chromium/issues/detail?id=1219139
+          if (
+            native &&
+            node.parentElement &&
+            window.getComputedStyle(node.parentElement).whiteSpace === 'pre'
```

Comment:
Probably unnecessary, but for some reason I think I’d feel better with:

```suggestion
            window?.getComputedStyle(node.parentElement)?.whiteSpace === 'pre'
```

Reply:
I don't think this will make any difference indeed but it doesn't hurt either so 🤷‍♂️